lib/Provider/future_provider.dart

```diff
@@ -163,18 +164,34 @@ class BrightnessNotifier extends AutoDisposeNotifier<Brightness>
     with WidgetsBindingObserver {
   @override
   void didChangePlatformBrightness() {
-    state = WidgetsBinding.instance.platformDispatcher.platformBrightness;
+    changeBrightness(
+        WidgetsBinding.instance.platformDispatcher.platformBrightness ==
+            Brightness.dark);
     super.didChangePlatformBrightness();
   }
 
   @override
   Brightness build() {
+    bool? isDark = Config().pref.getBool("isDark");
     WidgetsBinding.instance.addObserver(this);
-    return WidgetsBinding.instance.platformDispatcher.platformBrightness;
+    if (isDark == null) {
+      Brightness brightness =
```

Comment:
Isn't this an async operation?, just having that without await seems risky. I think it better if you can change this to a future provider.

Reply:
The reason i did not put any async operations is because adding brightness to shared preference has got nothing returning returning the value. 
If my operations depended adding values to the shared preference for the first time, then yes it makes sense to "await" the value being put inside the shared preference and then moving forward. However, in this case I don't exactly have to wait for operation to happen since I am not using the result of adding values to shared preference for the rest of the operations.

if `bool? isDark = Config().pref.getBool("isDark");` was an asynchronous operation, then yes, I would have to change the Notifier to async notifier since the rest of my operations depended upon the result from the getBool operation. Therefore I will have to "await" for that result of the getBool.

Reply:
The reason i did not put any async operations is because adding brightness to shared preference has got nothing returning returning the value.
If my operations depended adding values to the shared preference for the first time, then yes it makes sense to "await" the value being put inside the shared preference and then moving forward. However, in this case I don't exactly have to wait for operation to happen since I am not using the result of adding values to shared preference for the rest of the operations.

if bool? isDark = Config().pref.getBool("isDark"); was an asynchronous operation, then yes, I would have to change the Notifier to async notifier since the rest of my operations depended upon the result from the getBool operation. Therefore I will have to "await" for that result of the getBool.

Reply:
Hmm, makes sense. I think it is also why you made 2 different providers for todo list as well. One for operations on todos that interact with the database and one for displaying the todos on the main_screen. Then you had the 2 notifier interact with one another. Basically you decoupled the asynchronous and synchronous operations and made sure the one didn't depend on the other. This basically removed the need to do any "await" operations making it easier for maintaining the code

Reply:
Yes... 😄

Reply:
Yea sure. 👍